src/mod_muc_room.erl

```diff
@@ -1852,6 +1852,22 @@ add_new_user(From, Nick, Packet, StateData) ->
 		Nodes = get_subscription_nodes(Packet),
 		NewStateData =
 		      if not IsSubscribeRequest ->
+			      case mod_muc:check_create_roomid(ServerHost, Room) of
+			       true ->
+			      	 ServerHost = StateData#state.server_host,
+			      	 Room = StateData#state.room,
+			      	 Host = StateData#state.host,
+			      	 DefRoomOpts = gen_mod:get_module_opt(ServerHost, mod_muc, default_room_options, []),
+			      	 Access = gen_mod:get_module_opt(ServerHost, mod_muc, access, all),
+			      	 HistorySize = gen_mod:get_module_opt(ServerHost, mod_muc, history_size, 20),
+			      	 RoomShaper = gen_mod:get_module_opt(ServerHost, mod_muc, room_shaper, none),
+			      	 QueueType = gen_mod:get_module_opt(ServerHost, mod_muc, queue_type,
+			      		 ejabberd_config:default_queue_type(ServerHost)),
+			      	 RMod = gen_mod:ram_db_mod(ServerHost, ?MODULE),
+			      	 {ok, Pid} = mod_muc:start_new_room(
```

Comment:
Ahh we're hoping to hit the restore here?

Reply:
ya